ksql-common/src/main/java/io/confluent/ksql/serde/Delimiter.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.serde;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.errorprone.annotations.Immutable;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+@Immutable
+public final class Delimiter {
+
+  public final char delimiter;
+
+  public static Delimiter of(final char ch) {
+    return new Delimiter(ch);
+  }
+
+  public static Delimiter of(final String str) {
+    return new Delimiter(parse(str));
+  }
+
+  private Delimiter(final char delimiter) {
+    this.delimiter = delimiter;
+  }
+
+  public static Character parse(final String providedValueDelimiter) {
+    if (providedValueDelimiter == null) {
+      return null;
+    }
+    if (providedValueDelimiter.trim().isEmpty()) {
+      throw new IllegalArgumentException(
+          "Delimiter cannot be empty"
+              + System.lineSeparator()
+              + "Example valid value: ';'"
+      );
+    }
+    if (providedValueDelimiter.length() == 1) {
+      return providedValueDelimiter.charAt(0);
+    } else {
+      final Character delim = NAMED_DELIMITERS.get(providedValueDelimiter);
+      if (delim != null) {
+        return delim;
+      }
+      throw new IllegalArgumentException(
```

Comment:
ConfigException add this anyway - so if I do this it will appear twice.

Reply:
You're assuming it will always be used by something that wraps it in a `ConfigException`.

I'd include the value here and then build a `ConfigException` that didn't include the value, e.g.

```
   throw new ConfigException("Configuration " + name + " is invalid:" + e.getMessage()); 
```

Reply:
fixed